### ISSUE TITLE
Feature idea: add CPE/PURL as fields to `projects`

### ISSUE THREAD

**Issue by antonleviathan** [fedora-infra/anitya#1821]
While working with anitya to set up a bot to bump package versions for [[Stageˣ]](https://codeberg.org/stagex/stagex), I had the thought that it would be nice to have two additional fields which would help the users figure out if they are exposed to CVEs as well as construct SBOMs:
* `cpe`: [CPE](https://nvd.nist.gov/products/cpe)
* `purl`: [PURL](https://github.com/package-url/purl-spec/blob/master/README.rst)

The intuition came from the fact that the project mappings in anitya allow us to take project names and map them onto our desired name - but we have a shared point of reference where we can add metadata easily.

**Comment by arthurzam** [comment]
As a Gentoo developer, I think this might be nice and has great potential for sync work on generating SBOMs across distros.

For example in gentoo, we started not long ago to add CPE metadata to packages, but out progress is slow. Mapping gentoo packages to anitya was faster (thanks to github/pypi/... mappings), so if for example this is implemented, we could upload our information, download remote mapping, and enrich both ends. We more distors join this, we could all help each other quite fast.

**Comment by Zlopez** [comment]
This sounds like interesting idea. Could you specify how the two new fields should work and look?

**Comment by Zlopez** [comment]
It doesn't seem difficult to implement, just adding few new fields to UI and database.

**Comment by antonleviathan** [comment]
Yep it's straight forward. I would like to figure out if it's worthwhile to add PURL. Any thoughts on that @Zlopez? As of right now it looks like it may not be the thing we want but I'm not sure.

**Comment by Zlopez** [comment]
@antonleviathan I don't know much about purl to say if this will be helpful or not for release-monitoring.org users. CPE sounds like a good addition, I'm just not sure why we need  both `cpe_2_3` and `cpe_2_2`.

**Comment by antonleviathan** [comment]
I agree `cpe_2_3` is probably enough.

**Comment by antonleviathan** [comment]
Shall I put together a PR @Zlopez?

**Comment by Zlopez** [comment]
PRs are welcome :-) Feel free to put one together.

**Comment by antonleviathan** [comment]
Btw it looks like PURL would work well too. There is a `generic` tag that would address my concern:
https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#generic

**Comment by pombredanne** [comment]
@antonleviathan PURL maintainer here.
We are also working on a PURL type for C/C++ and packages that live off registries, like the kernel, this should land soon now, as we have consensus, including consulting with the kernel maintainers. Also of interest:
- PURL is now in the CVE schema https://github.com/CVEProject/cvelistV5/blob/1df9fb922e2c4fd162a8c0d63e44b7cc93f52f1f/README.md#L8
- also in SPDX, CycloneDX, OpenVEX and CSAF specs.

Tell me how I can help here.

**Comment by Zlopez** [comment]
@pombredanne I saw people just adding this as distro in release-monitoring.org, see https://release-monitoring.org/distro/CPE%20NVD%20NIST/.

I'm thinking about closing this ticket as it seems that the distro is enough, feel free to add one for PURL as well.
